components/org.wso2.carbon.identity.api.server.idp/org.wso2.carbon.identity.api.server.idp.v1/src/main/java/org/wso2/carbon/identity/api/server/idp/v1/core/ServerIdpManagementService.java

```diff
@@ -2839,34 +2856,89 @@ private ProvisioningConnectorConfig createProvisioningConnectorConfig(String out
      * @return Federated authenticator config of the specified ID.
      */
     private FederatedAuthenticatorConfig updateFederatedAuthenticatorConfig(String federatedAuthenticatorId,
-                                                                            FederatedAuthenticatorPUTRequest
-                                                                                    authenticator) {
-
-        FederatedAuthenticatorConfig authConfig = new FederatedAuthenticatorConfig();
-        String authenticatorName = base64URLDecode(federatedAuthenticatorId);
-        authConfig.setName(authenticatorName);
-        authConfig.setDisplayName(getDisplayNameOfAuthenticator(authenticatorName));
-        authConfig.setEnabled(authenticator.getIsEnabled());
+                  FederatedAuthenticatorPUTRequest authenticator) throws IdentityProviderManagementClientException {
 
+        String authenticatorName = getDecodedAuthName(federatedAuthenticatorId);
+        FederatedAuthenticatorConfig authConfig;
         String definedByType = null;
         if (authenticator.getDefinedBy() != null) {
             definedByType = authenticator.getDefinedBy().toString();
         }
-        authConfig.setDefinedByType(resolveDefinedByType(authenticatorName, definedByType, false));
+        definedByType = resolveDefinedByType(authenticatorName, definedByType, false).toString();
```

Comment:
Better not have a boolean passed to indicate if this an authenticator create or update flow.
Either break it into two or pass the respective operation as an input

Reply:
Addressed.